src/mount.rs

```diff
@@ -0,0 +1,84 @@
+use request::Request;
+use response::Response;
+use middleware::{Continue, Middleware, MiddlewareResult};
+
+use hyper::uri::RequestUri::AbsolutePath;
+
+pub struct Mount {
```

Comment:
Should be able to get static dispatch with the following:

```rust
pub struct Mount<M: Middleware + Send + Sync + 'static> {
    mount_point: String,
    middleware: M
}
```

(It may not be required to specify `+Send+Sync+'static` as they are implied by `Middleware` but I forget if that propagates)